File: frontend/src/types/employee.ts
export interface Employee {
    id: number;
    name: string;
    department: string;
    role: string;
    joiningDate: string;
    status: 'Active' | 'Archived';
    performanceScore: number;
    isArchived: boolean;
    createdAt: string;
    updatedAt: string;
}

export interface EmployeeDto{
    name: string;
    department: string;
    role: string;
    joiningDate: string;
    status: 'Active' | 'Archived';
    performanceScore: number;
}

export interface EmployeeFilter {
    searchTerm?: string;
    department?: string;
    status?: string;
    startDate?: string;
    endDate?: string;
    sortBy?: string;
    sortDescending?: boolean;
    page: number;
    pageSize: number;
}

File: backend/EmployeeDashboardApi/Models/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace EmployeeDashboardApi.Models
{
    public class Employee
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Department { get; set; } = string.Empty;

        [Required]
        public string Role { get; set; } = string.Empty;

        [DataType(DataType.Date)]
        public DateTime JoiningDate { get; set; }

        public string Status { get; set; } = "Active";
        public int PerformanceScore { get; set; } = 0;
        public bool IsArchived { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
